torch/fx/experimental/symbolic_shapes.py

```diff
@@ -401,6 +402,59 @@ def is_concrete_float(a: Union[float, SymFloat]) -> bool:
     return False
 
 
+def is_concrete_bool(a: Union[bool, SymBool]) -> bool:
+    """
+    Utility to check if underlying object
+    in SymBool is concrete value. Also returns
+    true if integer is passed in.
+
+    Args:
+        a (SymBool or bool): Object to test if it bool
+    """
+    assert isinstance(a, (SymBool, bool))
+
+    if isinstance(a, bool):
+        return True
+
+    if isinstance(
+        a.node.expr, (sympy.logic.boolalg.BooleanTrue, sympy.logic.boolalg.BooleanFalse)
+    ):
+        return True
+
+    return False
+
+
+def has_static_value(a: Union[SymBool, SymFloat, SymInt, bool, float, int]) -> bool:
+    """
+    User-code friendly utility to check if a value is static or dynamic.
+    Returns true if given a constant, or a symbolic expression with a fixed value.
+
+    Args:
+        a (Union[SymBool, SymFloat, SymInt, bool, float, int]): Object to test
+    """
+    assert isinstance(a, (SymBool, SymFloat, SymInt, bool, float, int))
+    if (
+        isinstance(a, (SymBool, bool))
+        and is_concrete_bool(a)  # type: ignore[arg-type]
+        or isinstance(a, (SymBool, float))
+        and is_concrete_float(a)  # type: ignore[arg-type]
+        or isinstance(a, (SymBool, int))
+        and is_concrete_int(a)  # type: ignore[arg-type]
+    ):
+        return True
+
+    if (
```

Comment:
I do not understand this part can you explain it and then add a comment to explain it?